src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnectorTask.java

```diff
@@ -514,7 +515,7 @@ public void commit() throws InterruptedException {
      * status whether this task is in the snapshot phase.
      */
     protected boolean isTaskInSnapshotPhase() {
-        return (this.coordinator == null) && this.coordinator.isSnapshotInProgress();
+        return (this.coordinator == null) || this.coordinator.isSnapshotInProgress();
```

Comment:
it will return true if coordinator is null is that what we want?
if (this.coordinator != null) && this.coordinator.isSnapshotInProgress()
in that case it will return false if coordinator is null

Reply:
Actually, coordinator being null means that the task is still initialisation phase and if we are saying that the connector has only two phases, i.e. snapshot and streaming then we can assume that the task is in streaming phase.

If coordinator is null, then we want the method to tell that we haven't moved on to the streaming phase so it is safe to return true.

If we change it to `if (this.coordinator != null) && this.coordinator.isSnapshotInProgress()` even this condition is okay but from a difference perspective. Consider that the coordinator is null, that is only in a situation when task is being initialised so `isTaskInSnapshotPhase()` will be `false`.